xml/object_test.go

```diff
@@ -0,0 +1,42 @@
+package xml
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestObject(t *testing.T) {
+	buffer := bytes.NewBuffer(nil)
+	scratch := make([]byte, 64)
+
+	object := newObject(buffer, &scratch)
+	object.Key("foo").String("bar")
+	object.Key("faz").String("baz")
+
+	e := []byte(`<foo>bar</foo><faz>baz</faz>`)
+	if a := buffer.Bytes(); bytes.Compare(e, a) != 0 {
+		t.Errorf("expected %+q, but got %+q", e, a)
+	}
+}
+
+func TestObjectWithNameSpaceAndAttributes(t *testing.T) {
+	buffer := bytes.NewBuffer(nil)
+	scratch := make([]byte, 64)
+
+	object := newObject(buffer, &scratch)
+
+	var metadata = func(t *TagMetadata) {
+		t.NamespacePrefix = "newspace"
+		t.NamespaceURI = "https://endpoint.com"
+		t.AttributeName = "attrName"
+		t.AttributeValue = "attrValue"
+	}
```

Comment:
Would be helpful to have tests for the optional attributeName behavior if that is intended to be supported.

Also want to document/clarify how nested namespaces should be used.